src/rgw/rgw_auth_s3.cc

```diff
@@ -447,7 +447,6 @@ int parse_v4_credentials(const req_info& info,                     /* in */
 			 boost::string_view& session_token,        /* out */
 			 const bool using_qs)                      /* in */
 {
-  dout(0) << "credential parse second time" << dendl;
```

Comment:
ok, so you cleaned up all kinds of things;  in a new feature, I think that the commits sent in the PR should be just the ones needed to add the feature in a logical way--not the history of the development in the author's checkout--the "cache cleanup" commits illustrate why

Reply:
Thanks @mattbenjamin .
I will squash and edit some of the commit msgs after fixing everything before merging the PR